rust/worker/src/compactor/compaction_manager.rs

```diff
@@ -349,6 +352,42 @@ impl Handler<Memberlist> for CompactionManager {
 
     async fn handle(&mut self, message: Memberlist, _ctx: &ComponentContext<CompactionManager>) {
         self.scheduler.set_memberlist(message);
+        if let Some(on_next_memberlist_update) = self.on_next_memberlist_signal.take() {
```

Comment:
a small amount of extra instrumentation so that the compactor probe waits for a memberlist to be set